### DISCUSSION TITLE
Get time remaining in cross buffer inside transition function.

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#2307] [Q&A] [answered]
Is there a way of getting the remaining length of the cross buffer inside the transition function.
```
def f(a, b)
add(normalize=false, [fade.out(a.source), fade.in(b.source)])
end
s = cross(duration={10.}, f, s)
```
In this example the remaining time in the buffer is 10. secs - one frame = 9.96s.

I have a few conditionals inside my transition function and wondered if there is a way of checking the remaining buffer time in respect to sequencing a small blank() in front of the b source fade in.

e.g.
```
add([fade.out(duration=t_length, a.source),sequence([blank(duration=space), fade.in(duration=t_length, b.source)])])
```
where space is the remaining buffer minus t_length.

At the moment i have a 10 sec transition length set using cross() which means the ```space = 9.96 - t_length```.

This has been fine but i feel there is a possibility that a gap is being introduced between a and b due to the buffer being slightly small than the 9.96s i predict from the crossfade log

**Comment by Russsgithub** [comment]
My problem here was related to having a large value from duration for the cross function (to get some long fades in certain circumstances) which I was trying to accommodate for in short xfade situations.

Using a function to set the duration of the cross function helped, so the transition has just over the fade length held in it buffer.

**Comment by toots** [comment, marked as answer]
I think should should be able to grab the `remaining()` value from either the ending or starting source.

**Comment by Russsgithub** [comment]
@toots Thanks, I'd not considered that.

On 24 Mar 2022 11:26, Romain Beauxis ***@***.***> wrote:

I think should should be able to grab the remaining() value from either the ending or starting source.
